recipes-extended/xen/files/libxl-xenmgr-support.patch

```diff
@@ -577,6 +579,18 @@ PATCHES
 +    return qmp_run_command(gc, domid, "quit", NULL, NULL, NULL);
 +}
 +
- int libxl__qmp_save(libxl__gc *gc, int domid, const char *filename, bool live)
+ int libxl__qmp_restore(libxl__gc *gc, int domid, const char *state_file)
  {
      libxl__json_object *args = NULL;
+@@ -1047,6 +1052,11 @@ static int qmp_change(libxl__gc *gc, lib
+     return rc;
+ }
+ 
++int libxl__qmp_stop(libxl__gc *gc, int domid)
++{
++    return qmp_run_command(gc, domid, "stop", NULL, NULL, NULL);
```

Comment:
If possible I'd like to get a second opinion on this; it looked okay to me, but I wasn't confident enough that there wouldn't be any undesired side effects.

Here's the original libxl__qmp_stop:
```
int libxl__qmp_stop(libxl__gc *gc, int domid)
{
    return qmp_run_command(gc, domid, "stop", NULL, NULL, NULL);
}
```

And here's the new suspend save:
```
void libxl__qmp_suspend_save(libxl__egc *egc,
                             libxl__domain_suspend_state *dsps)
{
    EGC_GC;
    int rc;
    libxl__ev_qmp *ev = &dsps->qmp;

    ev->ao = dsps->ao;
    ev->domid = dsps->domid;
    ev->callback = dm_stopped;
    ev->payload_fd = -1;

    rc = libxl__ev_qmp_send(gc, ev, "stop", NULL);
    if (rc)
        goto error;

    return;

error:
    dsps->callback_device_model_done(egc, dsps, rc);
}
```

Reply:
Re-introducing libxl__qmp_stop is probably the simplest change to keep the existing code working.  But it is re-introducing something that was removed upstream, so it will be maintenance going forward.